cantools/database/utils.py

```diff
@@ -297,6 +297,11 @@ def prune_signal_choices(signal):
 
     if i >= 0:
         full_prefix = full_prefix[0:i]
+    else:
+        # full_prefix does not contain an underscore
+        # but the following algorithm assumes it does
+        # and would strip too much
+        return
```

Comment:
ops. that's a bug indeed. Thanks for the fix! Question: should we just remove the common prefix? this would imply that your examples get reduced to "Err" and "OK", but depending on the conventions used, it might lead to unexpected results as well...